simple_cbor.py

```
@@ -665,6 +666,343 @@ def __len__(self):
+        """
+        keys = path.split(separator)
+        current = self.data
+        
+        # Navigate to parent
+        for key in keys[:-1]:
+            if isinstance(current, dict):
+                if key not in current:
+                    if create:
+                        current[key] = {}
+                    else:
+                        raise KeyError(f"Path not found: {key}")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)